tree/tree/src/TTree.cxx

```diff
@@ -6833,6 +6833,38 @@ bool TTree::MemoryFull(Int_t nbytes)
    return true;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Function merging branch structure from an outside tree into the current one
+///
+/// If the same branch name already exists in "this", it's skipped, only new ones
+/// are copied.
+/// Entries are not copied, just branch name / type is cloned
+/// Branches marked as DoNotProcess are not merged
+/// @param tree the outside TTree whose branches will be copied into this tree
+/// @return boolean true on sucess, false otherwise
+
+bool TTree::MergeBranches(TTree* tree)
+{
+   if (tree) {
+      TObjArray* newbranches = tree->GetListOfBranches();
+      Int_t nbranches = newbranches->GetEntriesFast();
+      for (Int_t i = 0; i < nbranches; ++i) {
+         TBranch* nbranch = (TBranch*) newbranches->UncheckedAt(i);
+         if (nbranch->TestBit(kDoNotProcess)) {
+            continue;
+         }
+         if (!FindBranch(nbranch->GetName())) {
```

Comment:
```suggestion
         if (!GetListOfBranches()->FindObject(nbranch->GetName())
```
as we really only want to look at top level branches and have an exact name match.